lib/charliecloud.py

```diff
@@ -446,6 +447,18 @@ def set_(dst_key, *src_keys):
             except AttributeError:
                FATAL("can't parse config: bad Env line: %s" % line)
             self.metadata["env"][k] = v
+      # History. Set the empty_layer value of each non-final history entry to
+      # true.
+      try:
+         hist = config["history"]
+      except AttributeError:
+         FATAL("config missing key 'history'")
+      hist = self.metadata_empty_layer_history(hist)
+      try:
+         hist[-1].pop("empty_layer")
+      except (IndexError, KeyError):
+         pass
```

Comment:
This can probably be eliminated by leaving the `empty_layer` workaround until push time.